src/scenarios/shared/runner.py

```diff
@@ -319,6 +340,87 @@ def run(self):
             startup.runtests(self.traits)
 
 
+        elif self.testtype == const.DEVICEMEMORYCONSUMPTION and self.devicetype == 'android':
+            getLogger().info("Clearing potential previous run nettraces")
+            for file in glob.glob(os.path.join(const.TRACEDIR, 'PerfTest', 'runoutput.trace')):
+                if exists(file):   
+                    getLogger().info("Removed: " + os.path.join(const.TRACEDIR, file))
+                    os.remove(file)
+
+            androidHelper = AndroidHelper()
+            try:
+                androidHelper.setup_device(self.packagename, self.packagepath, self.animationsdisabled)
+
+                # Create the fullydrawn command
+                clearProcStatsCmd = [ 
+                    androidHelper.adbpath,
+                    'shell',
+                    'dumpsys',
+                    'procstats',
+                    '--clear'
+                ]
+
+                captureProcStatsCmd = [ 
+                    androidHelper.adbpath,
+                    'shell',
+                    'dumpsys',
+                    'procstats',
+                    self.packagename,
+                    '--section',
+                    'proc'
+                ]
+
+                clearLogsCmd = [
+                    androidHelper.adbpath,
+                    'logcat',
+                    '-c'
+                ]
+
+                allResults = []
+                for i in range(self.testiterations):
```

Comment:
I know that this is how we currently do this for startup, but why can't we move the execution of the remote commands into the memory consumption tool, and possibly also bring that back for the startup side. I honestly can't remember if we took this approach last time out of convenance or if there was a technical reason why we had to use that approach.

Reply:
After thinking about this, I think at least part of the reason for having the remote commands in the python instead of the tool is that the tools are currently setup to run processes rather than execute a set of steps (hopefully that makes sense). I am not set in having the execution stuff live in python over the tool, I think it is just a matter of figuring out if we want to support steps outside of executing an exe to live in the tools.

Reply:
One potential solution to this would be to move the runner.py section for the startup and device memory to a different script that we call in these specific cases.